MarsRover.Tests/MapParserShould.cs

```diff
@@ -1,45 +1,26 @@
 using System;
 using System.Collections.Generic;
-using System.Linq;
-using Moq;
+using FluentAssertions;
 using Xunit;
 
 namespace MarsRover.Tests
 {
     public class MapParserShould
     {
-        [Fact]
-        public void ReturnMap_GivenValidString()
+        [Theory]
+        [MemberData(nameof(MapParserShouldTestData.MapParserTestData), MemberType = typeof(MapParserShouldTestData))]
+        public void ReturnMap_GivenValidString(List<Square> squares, int width, int height, int expectedObstacleCount, int expectedSquareCount, params string[] inputs)
         {
-            var squares = new List<Square>
-            {
-                new Square(SquareContent.Obstacle, 0, 0),
-                new Square(SquareContent.None, 1, 0),
-                new Square(SquareContent.None, 2, 0),
-                new Square(SquareContent.None, 3, 0),
-                new Square(SquareContent.None, 0, 1),
-                new Square(SquareContent.None, 1, 1),
-                new Square(SquareContent.None, 2, 1),
-                new Square(SquareContent.None, 3, 1),
-                new Square(SquareContent.None, 0, 2),
-                new Square(SquareContent.None, 1, 2),
-                new Square(SquareContent.None, 2, 2),
-                new Square(SquareContent.None, 3, 2)
-            };
-            var expected = new Map(4, 3, squares);
-            var expectedCount = 1;
-            var actualCount = 0;
+            var mapString = String.Join(Environment.NewLine, inputs);
+            var expected = new Map(width, height, squares);
 
-            var result = MapParser.ParseMap("ONNN\nNNNN\nNNNN");
-            foreach(var square in result.Squares)
-            {
-                if (square.HasObstacle()) actualCount++;
-            }
-
-            Assert.Equal(12, result.Squares.Count());
+            var result = MapParser.ParseMap(mapString);
+            var actualObstacleCount = result.Squares.FindAll(c => c.Content == SquareContent.Obstacle).Count;
+            
+            result.Squares.Should().HaveCount(expectedSquareCount);
+            result.Should().Equals(expected);
+            Assert.Equal(expectedObstacleCount, actualObstacleCount);

```

Comment:
I think FluentAssertions checks the objects for equality several layers deep, so it will check that your squares are equal as well as the content of the squares (to be sure - try changing one of the square contents in your test data and see if the test fails when expected). Since the type of content and the contents of the squares are already covered by fluent assertions, is there much value in asserting that the square and obstacle count are equal? Same goes for the type check.